app/src/main/java/com/rootstrap/android/ui/base/BaseFragment.kt

```diff
@@ -16,4 +22,36 @@ open class BaseFragment : Fragment(), BaseView {
     override fun showError(message: String?) {
         LoadingDialogUtil.showError(message, requireContext())
     }
+
+    fun openActivity(activity: Class<*>, clearTask: Boolean = false, extras: Bundle? = null) {
+        requireActivity().startActivity(
+            Intent(requireActivity(), activity).also {
+                if (clearTask) {
+                    it.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                    it.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
+                }
+                extras?.let { bundle -> it.putExtras(bundle) }
+            }
+        )
+        if (clearTask) {
+            requireActivity().finish()
+        }
+    }
+
+    fun observeNetwork(baseViewModel: BaseViewModel) {
+        baseViewModel.networkState.observe(this, {
+            when (it) {
+                NetworkState.loading -> showProgress()
+                NetworkState.idle -> hideProgress()
+                else -> showError(baseViewModel.error ?: getString(R.string.default_error))
+            }
+        })
+    }
+
+    fun navigateTo(routeOrAction: Int, bundle: Bundle? = null) {
+        requireActivity().takeIf { it is BaseNavActivity }?.let {
+            it as BaseNavActivity
+            it.navigateTo(routeOrAction, bundle)
+        }
+    }
```

Comment:
Wdyt about using SafeArgs's NavDirections here instead of an int? 
Also we do not need to call the parent activity to navigate, as we can use the findNavController() extension from navigation-fragment-ktx

Reply:
Since the fragments are under the activity, I think that the activity should be the one in charge of the navigation, in case that you want to track it or do something else with it.